python/cuml/cuml/tests/experimental/accel/test_basic_estimators.py

```diff
@@ -172,6 +172,59 @@ def test_proxy_facade():
 
             assert original_value == proxy_value
 
+            
+def test_proxy_clone():
+    # Test that cloning a proxy estimator preserves parameters, even those we
+    # translate for the cuml class
+    pca = PCA(n_components=42, svd_solver="arpack")
+    pca_clone = clone(pca)
+
+    assert pca.get_params() == pca_clone.get_params()
+
+
+def test_proxy_params():
+    # Test that parameters match between constructor and get_params()
+    # Mix of default and non-default values
+    pca = PCA(
+        n_components=5,
+        copy=False,
+        # Pass in an argument and set it to its default value
+        whiten=False,
+    )
+
+    params = pca.get_params()
+    assert params["n_components"] == 5
+    assert params["copy"] is False
+    assert params["whiten"] is False
+    # A parameter we never touched, should be the default
+    assert params["tol"] == 0.0
+
+    # Check that get_params doesn't return any unexpected parameters
+    expected_params = set(
+        [
+            "n_components",
+            "copy",
+            "whiten",
+            "tol",
+            "svd_solver",
+            "n_oversamples",
+            "random_state",
+            "iterated_power",
+            "power_iteration_normalizer",
+        ]
+    )
+    assert set(params.keys()) == expected_params
+
+
+def test_roundtrip():
+    import cuml
+    from sklearn import cluster
+
+    km = cluster.KMeans(n_clusters=13)
+    ckm = cuml.KMeans.from_sklearn(km)
+
+    assert ckm.n_clusters == 13
```

Comment:
Lucky number 13 :D

Reply:
I think we should merge this PR. It improves things and fixes several things.

We can keep improving the `from_`/`as_sklearn` round tripping. I think the test from https://github.com/rapidsai/cuml/pull/6342/files#r1963552769 still doesn't pass (even if you exclude the raft handle). But lets look at that in a new PR